compose-ktx/src/main/kotlin/modifier/ApplyWhen.kt

```diff
@@ -0,0 +1,32 @@
+package com.redmadrobot.extensions.compose.modifier
+
+import androidx.compose.ui.Modifier
+
+/**
+ * Applies the given block of modifications to the Modifier if the condition is true.
+ *
+ * @param condition condition to determine if the block should be applied.
+ * @param block Lambda function that modifies the Modifier.
+ * @return Modified Modifier based on the condition.
+ */
+public inline fun Modifier.applyWhen(
```

Comment:
It will be better to name this function `applyIf` to be consistent with naming used in Kotlin stdlib.
However, it looks like this one could be an extension on any type. And maybe it should reside in some sort of `kotlin-ktx` :D (not sure about naming)

```kotlin
inline fun <T> T.applyIf(confition: Boolean, block: T.() -> T): T
```

I remembered that we have had more extensions that could be added to this possible `kotlin-ktx` module.

Reply:
Renamed it - but kept the package the same and did not make it generic.

Kotlin-ktx - i suggest to decide its fate after discussing compose-ktx (if we decide to add kotlin-ktx - I will bring it as a separate pr).

Reply:
Continue here: https://github.com/RedMadRobot/gears-android/pull/52#discussion_r1611513356
And here: https://github.com/RedMadRobot/gears-android/pull/52#discussion_r1611570604